src/MLJXGBoostInterface.jl

```diff
@@ -1,770 +1,212 @@
-    ,normalize_type="tree"
-    ,rate_drop=0.0
-    ,one_drop=0
-    ,skip_drop=0.0
-    ,feature_selector="cyclic"
-    ,top_k=0
-    ,tweedie_variance_power=1.5
-    ,objective="count:poisson"
-    ,base_score=0.5
-    ,eval_metric="rmse"
-    ,seed=0
-    ,nthread = 1)
-
-    model = XGBoostCount(
-    num_round
-    ,booster
-    ,disable_default_eval_metric
-    ,eta
-    ,gamma
-    ,max_depth
-    ,min_child_weight
-    ,max_delta_step
-    ,subsample
-    ,colsample_bytree
-    ,colsample_bylevel
-    ,lambda
-    ,alpha
-    ,tree_method
-    ,sketch_eps
-    ,scale_pos_weight
-    ,updater
-    ,refresh_leaf
-    ,process_type
-    ,grow_policy
-    ,max_leaves
-    ,max_bin
-    ,predictor #> gpu version not currently working with Julia, maybe remove completely?
-    ,sample_type
-    ,normalize_type
-    ,rate_drop
-    ,one_drop
-    ,skip_drop
-    ,feature_selector
-    ,top_k
-    ,tweedie_variance_power
-    ,objective
-    ,base_score
-    ,eval_metric
-    ,seed
-    ,nthread)
-
-     message = MMI.clean!(model)
-     isempty(message) || @warn message
-
-    return model
-end
-
-function MMI.clean!(model::XGBoostCount)
-    warning = ""
-    if(!(model.objective in ["count:poisson"]))
-        warning *= "Changing objective to \"poisson\", "*
-                       "the only supported value. "
-        model.objective="poisson"
+function MMI.feature_importances(model::XGTypes, (booster, _), (features,))
+    dict = XGB.importance(booster, model.importance_type)
+    if length(last(first(importance_dict))) > 1
+        [features[k] => zero(first(v)) for (k, v) in dict]
+    else
+        [features[k] => first(v) for (k, v) in dict]
     end
-    return warning
 end
 
-function MMI.fit(model::XGBoostCount
-             , verbosity::Int     #> must be here even if unsupported in pkg
-             , X
-             , y)
-
-    silent = verbosity > 0 ?  false : true
-
-    Xmatrix = _to_array(MMI.matrix(X))
-    dm = XGBoost.DMatrix(Xmatrix, label=_to_array(y))
-
-    seed =
-        model.seed == -1 ? generate_seed() : model.seed
-
-    fitresult = XGBoost.xgboost(dm, model.num_round;
-                                kwargs(model, silent, seed, "count:poisson",model.eval_metric)...)
-    features = schema(X).names
-    importances = [named_importance(fi, features) for
-                   fi in XGBoost.importance(fitresult)]
-
-    cache = nothing
-    report = (feature_importances=importances, )
-
-    return fitresult, cache, report
-
+function _feature_names(X, dmatrix)
+    schema = Tables.schema(X)
+     if schema === nothing
+        features = [Symbol("x$j") for j in 1:ncols(dmatrix)]
+    else
+        features = schema.names |> collect
+    end
+    return features
 end
 
-function MMI.predict(model::XGBoostCount
-        , fitresult
-        , Xnew)
-    Xmatrix = _to_array(MMI.matrix(Xnew))
-    return XGBoost.predict(fitresult, Xmatrix)
+function MMI.fit(model::XGBoostAbstractRegressor, verbosity::Integer, X, y)
```

Comment:
```suggestion
function MMI.fit(model::XGBoostAbstractRegressor, verbosity, X, y)
```

Typing of `verbosity` is now an anachronism  and not recommended.